isisd/isis_adjacency.c

```diff
@@ -121,6 +122,50 @@ struct isis_adjacency *isis_adj_lookup_snpa(const uint8_t *ssnpa,
 	return NULL;
 }
 
+bool isis_adj_exists(const struct isis_area *area, const uint8_t *sysid)
+{
+	struct isis_circuit *circuit;
+	struct listnode *node;
+
+	for (ALL_LIST_ELEMENTS_RO(area->circuit_list, node, circuit)) {
+		struct isis_adjacency *adj;
+		struct listnode *anode;
+		struct list *adjdb;
+
+		switch (circuit->circ_type) {
+		case CIRCUIT_T_BROADCAST:
+			for (int level = IS_LEVEL_1; level <= IS_LEVEL_2;
+			     level++) {
+				if ((area->is_type & level) == 0)
+					continue;
```

Comment:
I'm a little confused by this line -- what does this do?

Reply:
It's a common idiom used in the isisd code to handle L1/L2 areas like both L1 and L2 areas (which is what they are).

Please take a look below:
```c
#define IS_LEVEL_1       1
#define IS_LEVEL_2       2
#define IS_LEVEL_1_AND_2 3
```

`IS_LEVEL_1_AND_2` was conveniently set to `3` so that `(IS_LEVEL_1_AND_2 & IS_LEVEL_1)` and `(IS_LEVEL_1_AND_2 & IS_LEVEL_2)` will be different from zero.